summingbird-storm-test/src/test/scala/com/twitter/summingbird/storm/StormTestUtils.scala

```diff
@@ -0,0 +1,183 @@
+import com.twitter.summingbird.storm.Storm.toStormSource
+import com.twitter.summingbird.storm.spout.TraversableSpout
+import com.twitter.summingbird._
+import com.twitter.util.Future
+import org.scalacheck.rng.Seed
+import org.scalacheck.{ Arbitrary, Cogen, Gen }
+
+import scala.collection.mutable
+
+object StormTestUtils {
+  def testStormEqualToMemory(producerCreator: ProducerCreator)(implicit storm: Storm): Unit = {
+    testStormEqualToMemory(producerCreator, Seed.random(), Gen.Parameters.default)
+  }
+
+  def testStormEqualToMemory(producerCreator: ProducerCreator, seed: Seed, params: Gen.Parameters)(implicit storm: Storm): Unit = {
+    val memorySourceCreator = new MemorySourceCreator()
+    val memoryStoreCreator = new MemoryStoreCreator(seed, params)
+    val memorySinkCreator = new MemorySinkCreator()
+    val memoryServiceCreator = new MemoryServiceCreator()
+    val memoryCtx = new CreatorCtx(memorySourceCreator, memoryStoreCreator, memorySinkCreator, memoryServiceCreator)
+
+    val stormSourceCreator = new StormSourceCreator()
+    val stormStoreCreator = new StormStoreCreator(seed, params)
+    val stormSinkCreator = new StormSinkCreator()
+    val stormServiceCreator = new StormServiceCreator
+    val stormCtx = new CreatorCtx(stormSourceCreator, stormStoreCreator, stormSinkCreator, stormServiceCreator)
+
+    val memory = new Memory()
+    memory.run(memory.plan(producerCreator.apply(memoryCtx)))
+
+//    assert(OnlinePlan(tail).nodes.size < 10)
+    StormTestRun(producerCreator.apply(stormCtx))
+
+    assertEquiv(memoryStoreCreator.ids(), stormStoreCreator.ids())
+    memoryStoreCreator.ids().foreach(id =>
+      assertEquiv(memoryStoreCreator.get(id), stormStoreCreator.get(id))
+    )
+    assertEquiv(memorySinkCreator.get(), stormSinkCreator.get())
+    stormSinkCreator.clear()
+  }
+
+  def sample[T: Arbitrary](seed: Seed, params: Gen.Parameters, id: String): T = {
+    implicitly[Arbitrary[T]].arbitrary(
+      params,
+      implicitly[Cogen[String]].perturb(seed, id)
+    ).get
+  }
+
+  def assertEquiv[T](expected: T, returned: T)(implicit equiv: Equiv[T]): Unit = {
+    assert(equiv.equiv(expected, returned), (expected.toString, returned.toString))
+  }
+}
+
+trait ProducerCreator {
+  def apply[P <: Platform[P]](ctx: CreatorCtx[P]): TailProducer[P, Any]
+}
+
+class CreatorCtx[P <: Platform[P]](
+  sourceCreator: SourceCreator[P],
+  storeCreator: StoreCreator[P],
+  sinkCreator: SinkCreator[P],
+  serviceCreator: ServiceCreator[P]
+) {
+  def source[T](data: List[T]): Source[P, T] = sourceCreator(data)
+  def store[K: Arbitrary, V: Arbitrary: Semigroup](id: String): P#Store[K, V] = storeCreator(id)
+  def sink[V: Ordering](id: String): P#Sink[V] = sinkCreator(id)
+  def service[K, V](fn: K => Option[V]): P#Service[K, V] = serviceCreator(fn)
+}
+
+trait SourceCreator[P <: Platform[P]] {
+  // Should be with time
```

Comment:
Can you make this comment more clear, I'm not sure what it means in this context.